### PR TITLE
fix(style): adding style to the apps detail table (AEROGEAR-8766)

### PR DESCRIPTION
**Pull request by laurafitzgerald**
## Motivation
https://issues.jboss.org/browse/AEROGEAR-8766
 
## What
Styling
- The table
- I also improved the blank slate somewhat

## Why
It's nice when things look nice

## How
css magic

## Verification Steps

1. Run `make serve`
2. UI should appear as below

Screenshot 2019-03-11 at 13 18 21

## Checklist:

- [x] Code has been tested locally by PR requester
- [ ] Changes have been successfully verified by another team member 

## Additional Notes

Pushing this as a WIP as I'm basing it on the styling as outlined in the story flat but I did make some additions that weren't asked for so happy to remove those if necessary.

### CODE REVIEW

File: ui/src/components/LandingPage.css [hunk 1]
```diff
@@ -0,0 +1,11 @@
+.title {
+    text-align: center;
```
Comment (damienomurchu) [inline ui/src/components/LandingPage.css]:
```suggestion
    text-align: center;
    text-align: left;
```

File: ui/src/components/Table.css [hunk 2]
```diff
@@ -0,0 +1,12 @@
+.apps-table {
```
Comment (damienomurchu) [inline ui/src/components/Table.css]: @laurafitzgerald can we increase the size of the table head font?

Comment (laurafitzgerald) [reply ui/src/components/Table.css]: @damienomurchu sure. Should I make it lower case then?

File: ui/src/components/LandingPage.css (hunk 1 above)
Comment (laurafitzgerald) [reply ui/src/components/LandingPage.css]: I didn't like it aligned to the left. But it is aligned left in the flat so I will do that.

File: ui/src/components/LandingPage.js [hunk 3]
```diff
@@ -12,6 +14,8 @@ class LandingPage extends React.Component {
     return (
       <>
         <Header />
+        <Title className="title" size="4xl">Mobile Apps</Title>
```
Comment (damienomurchu) [inline ui/src/components/LandingPage.js]:
```suggestion
        <Title className="title" size="2xl">Mobile Apps</Title>

```

Comment (damienomurchu) [reply ui/src/components/LandingPage.js]: `4x1` seems larger than Title font in flat

File: ui/src/components/common/Header.js [hunk 4]
```diff
@@ -19,7 +20,9 @@ import { connect } from 'react-redux';
 import { toggleHeaderDropdown } from '../../actions/actions-ui';
 
 class Header extends React.Component {
-  onTitleClick = () => {};
+  onTitleClick () {
+    this.props.history.push('/');
```
Comment (damienomurchu) [inline ui/src/components/common/Header.js]: Are we using this navigation history/ is it wired into React already?

Comment (laurafitzgerald) [reply ui/src/components/common/Header.js]: It uses Router. the history objection tracks the navigation across all components. It works really well. You can check it by navigating between the home page and the two apps and then using the back button.

File: ui/src/components/LandingPage.js (hunk 3 above)
Comment (laurafitzgerald) [reply ui/src/components/LandingPage.js]: I tried out the 2xl. Went with 3xl in the end as it seemed to look the nicer of the 3 options.

**Comment by laurafitzgerald** [comment]
@damienomurchu @craicoverflow I've pushed some new changes which add some fixes other than the styling. There's a method and action name change and also some fixes on the navigation and some state setting which was a little broken too. I also updated tests alongside those fixes. Would you mind reviewing again when you get a chance.

File: ui/src/components/Table.css (hunk 2 above)
Comment (laurafitzgerald) [reply ui/src/components/Table.css]: I added some styling to the table header and I think it makes it stand out now without increasing the font.

**Comment by damienomurchu** [comment]
@laurafitzgerald will review shortly. One q - added css files don't match the component file name they're being used with, is this deliberate?

File: ui/src/containers/AppsTableContainer.js [hunk 5]
```diff
@@ -15,22 +16,22 @@ export class AppsTableContainer extends React.Component {
   componentWillMount () {
     this.props.getApps();
   }
-  onRowClick (event, rowId, props) {
+  onRowClick (_event, rowId) {
```
Comment (damienomurchu) [inline ui/src/containers/AppsTableContainer.js]: Is `_event` param required (as it's not being used) ?

File: ui/src/containers/AppsTableContainer.js [hunk 6]
```diff
@@ -15,22 +16,22 @@ export class AppsTableContainer extends React.Component {
...
     var app = this.props.apps.data.filter(app => {
       return app.appName === rowId[0];
     });
     const id = app[0].id;
     const path = '/app/' + id;
     this.props.history.push(path);
   }
-  onSort (_event, index) {
-    this.props.reverseAppsTableSort(index);
+  onSort (_event, index, direction) {
```
Comment (damienomurchu) [inline ui/src/containers/AppsTableContainer.js]: Is `_event` param required (as it's not being used)

File: ui/src/actions/__tests__/actions-ui.test.js [hunk 7]
```diff
@@ -1,15 +1,18 @@
 import * as actions from '../actions-ui';
-import { REVERSE_SORT } from '../types';
+import { APPS_SORT } from '../types';
```
Comment (damienomurchu) [inline ui/src/actions/__tests__/actions-ui.test.js]: I wonder is this descriptive enough to distinguish it from the sort we will be doing on the the AppDetails table, or do you have a suggestion on what the sort action for AppDetails should be named so we keep them consistent?

File: ui/src/containers/AppsTableContainer.js (hunk 5 above)
Comment (craicoverflow) [reply ui/src/containers/AppsTableContainer.js]: It is the first parameter in a default function, it cannot be removed.

File: ui/src/components/AppsTable.js [hunk 8]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
 import { Table, TableHeader, TableBody } from '@patternfly/react-table';
 import { withRouter } from 'react-router-dom';
+import './Table.css';
```
Comment (damienomurchu) [inline ui/src/components/AppsTable.js]: Should this be `AppsTable.css` ?

File: ui/src/containers/AppsTableContainer.js (hunk 6 above)
Comment (craicoverflow) [reply ui/src/containers/AppsTableContainer.js]: It is the first parameter in a default function, it cannot be removed.

File: ui/src/actions/__tests__/actions-ui.test.js (hunk 7 above)
Comment (laurafitzgerald) [reply ui/src/actions/__tests__/actions-ui.test.js]: I would suggest APP_SORT for the app table

File: ui/src/components/AppsTable.js (hunk 8 above)
Comment (laurafitzgerald) [reply ui/src/components/AppsTable.js]: I was thinking we could use it for both tables as we probably want them to be consistent which is why I left the name more generic.

File: ui/src/components/common/Header.js [hunk 9]
```diff
@@ -18,8 +19,17 @@ import PropTypes from 'prop-types';
...
-  onTitleClick = () => {};
+export class Header extends React.Component {
+  constructor (props) {
+    super(props);
+    this.onTitleClick = this.onTitleClick.bind(this);
+    this.onUserDropdownToggle = this.onUserDropdownToggle.bind(this);
+    this.onUserDropdownSelect = this.onUserDropdownSelect.bind(this);
+    this.onLogoutUser = this.onLogoutUser.bind(this);
+  }
+  onTitleClick () {
```
Comment (damienomurchu) [inline ui/src/components/common/Header.js]:
```suggestion
  onTitleClick = () => {
```

Comment (damienomurchu) [reply ui/src/components/common/Header.js]: Converting to arrow function as suggested will remove need to bind `this` to the method in the constructor

File: ui/src/components/common/Header.js [hunk 10]
```diff
@@ -18,8 +19,17 @@ import PropTypes from 'prop-types';
...
 
-class Header extends React.Component {
-  onTitleClick = () => {};
+export class Header extends React.Component {
+  constructor (props) {
+    super(props);
+    this.onTitleClick = this.onTitleClick.bind(this);
+    this.onUserDropdownToggle = this.onUserDropdownToggle.bind(this);
+    this.onUserDropdownSelect = this.onUserDropdownSelect.bind(this);
+    this.onLogoutUser = this.onLogoutUser.bind(this);
```
Comment (damienomurchu) [inline ui/src/components/common/Header.js]: The `this` binding being done here is not necessary as `this` is being bound below by using arrow functions to define the handlers. 

If all `bind` calls are removed, this constructor can be eliminated too

File: ui/src/components/common/Header.js [hunk 11]
```diff
@@ -60,7 +66,7 @@ class Header extends React.Component {
       </Toolbar>
     );
 
-    const Header = <PageHeader logo="MOBILE SECURITY SERVICE" logoProps={logoProps} toolbar={toolbar} />;
+    const Header = <PageHeader logo="MOBILE SECURITY SERVICE" onClick={this.onTitleClick} toolbar={toolbar} />;
```
Comment (damienomurchu) [inline ui/src/components/common/Header.js]: Is `onClick` supported by the PageHeader component? (Documentation seems to indicate it isn't - http://patternfly-react.surge.sh/patternfly-4/components/page)

Comment (laurafitzgerald) [reply ui/src/components/common/Header.js]: I believe it's just a property. I was being passed as a property in the previous implementation also.

File: ui/src/reducers/index.js [hunk 12]
```diff
@@ -25,19 +25,19 @@ const initialState = {
 
 export default (state = initialState, action) => {
   switch (action.type) {
-    case REVERSE_SORT:
-      const reversedOrder = state.sortBy.direction === SortByDirection.asc ? SortByDirection.desc : SortByDirection.asc;
+    case APPS_SORT:
       const index = action.payload.index;
-      const sortedRows = state.apps.rows.sort((a, b) => (a[index] < b[index] ? -1 : a[index] > b[index] ? 1 : 0));
-      const sortedApps = reversedOrder === SortByDirection.asc ? sortedRows : sortedRows.reverse();
+      var sortedRows = state.apps.rows.sort((a, b) => (a[index] < b[index] ? -1 : a[index] > b[index] ? 1 : 0));
```
Comment (damienomurchu) [inline ui/src/reducers/index.js]: Change from var to let here

File: ui/src/components/common/Header.js (hunk 11 above)
Comment (damienomurchu) [reply ui/src/components/common/Header.js]: Can you share where it was being passed previously? 

I haven't ran the code to check if the onClick call works here - do you know if it does?

Comment (laurafitzgerald) [reply ui/src/components/common/Header.js]: It's in the same file. I removed a few lines above this.

**Comment by craicoverflow** [comment]
Header does not stay full width when viewed on a smaller screen. This was also happening before this PR (switched back to master to check) but now the gap has grown significantly so something here may be affecting it.

Screenshot from 2019-03-12 09-46-53

File: ui/src/components/common/Header.js (hunk 11 above)
Comment (damienomurchu) [reply ui/src/components/common/Header.js]: Can you confirm the `onClick()` still works, as before it was being passed in as part of the object to the logoProps attribute to PageHeader. PageHeader doesn't seem to accept an `onClick` property[1], and looks like an onClick handler can only be passed in via logoProps to PageHeader

[1]: https://github.com/patternfly/patternfly-react/blob/0b637feaf467a860ee69263bb2c477a3d0090c22/packages/patternfly-4/react-core/src/components/Page/PageHeader.js#L8

Comment (laurafitzgerald) [reply ui/src/components/common/Header.js]: I can confirm that it's working.

File: ui/src/components/LandingPage.css [hunk 13]
```diff
@@ -0,0 +1,5 @@
+.title {
+    padding-top: .5em;
+    margin-bottom: .5em;
+    margin-left: 2em;
```
Comment (damienomurchu) [inline ui/src/components/LandingPage.css]:
```suggestion
    margin-left: 1em;
```
With `2em` the title looks too indented - `1em` aligns the header text, title text, and table text, wdyt?

File: ui/src/components/common/Header.js (hunk 11 above)
Comment (damienomurchu) [reply ui/src/components/common/Header.js]: Me too, resolving this now

**Comment by damienomurchu** [comment]
@laurafitzgerald, PR looks good, just one comment left re left padding of Title

File: ui/src/components/LandingPage.css (hunk 13 above)
Comment (laurafitzgerald) [reply ui/src/components/LandingPage.css]: When I use 1em I see this: 

Screenshot 2019-03-12 at 11 55 44

Could you post a screenshot also?

Comment (damienomurchu) [reply ui/src/components/LandingPage.css]: Screenshot from 2019-03-12 11-59-53
Screenshot from 2019-03-12 12-00-06

Comment (damienomurchu) [reply ui/src/components/LandingPage.css]: With `2em`:
Screenshot from 2019-03-12 12-02-53
Screenshot from 2019-03-12 12-03-09

Comment (damienomurchu) [reply ui/src/components/LandingPage.css]: Screenshot from 2019-03-12 12-04-10

**Comment by laurafitzgerald** [comment]
@craicoverflow the responsive header issue should be sorted with the new commit. Would you mind verifying that when you get a chance.

File: ui/src/components/LandingPage.css (hunk 13 above)
Comment (laurafitzgerald) [reply ui/src/components/LandingPage.css]: I'm using Chrome: Version 72.0.3626.109. Strange that I have exactly the same and it looks very different. Would the version of Chrome make that much difference?

Comment (laurafitzgerald) [reply ui/src/components/LandingPage.css]: @damienomurchu pushed a new change which should resolve this. Would you mind verifying?

File: ui/src/components/LandingPage.css [hunk 14]
```diff
@@ -1,5 +1,5 @@
 .title {
-    padding-top: .5em;
-    margin-bottom: .5em;
-    margin-left: 2em;
+    margin-left: 5%;
```
Comment (damienomurchu) [inline ui/src/components/LandingPage.css]: @laurafitzgerald is there a reason for the change in units used for the styling?

Comment (laurafitzgerald) [reply ui/src/components/LandingPage.css]: yes, i believe we'll have less inconsistency. When I set both the table and the Header to have 2em they didn't line up. When using percentage they will line up.

Comment (laurafitzgerald) [reply ui/src/components/LandingPage.css]: @damienomurchu does it fix the discrepancy on your end?

**Comment by damienomurchu** [comment]
@laurafitzgerald reviewed. Fix for overflowing table is probably the best we can hope for with the present implementation of the table in PF4 which does not appear to be as responsive as it should be compared to patternfly-core components. 

Just had 1 question left in comments. Think you addressed everything else

File: ui/src/components/LandingPage.css (hunk 14 above)
Comment (damienomurchu) [reply ui/src/components/LandingPage.css]: Understand. Ideally I think we would look at using `rem` (which is used by PF, and the spacing for the components specified usually in `rem` at the bottom of each component page). If we need to get this merged in due to time constraints we can go with `%` but I think `rem` might be better (but we can refactor in future).

Comment (damienomurchu) [reply ui/src/components/LandingPage.css]: Left comment above

**Pull request merged**